query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java

```diff
@@ -245,6 +246,9 @@ public void onFilterResult(Object userContext, Object instance, Object eventType
                event = clone;
             }
             for (DelegatingCacheEntryListenerInvocation<K, V> invocation : invocations) {
+               if (event.isPre() && invocation.getObservation() == Listener.Observation.POST) {
```

Comment:
For completeness do we want the inverse? When event is post and observation is PRE ?